lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -46,17 +46,48 @@ class PublicLayoutHelper
       attr_reader :footer_navigation, :footer_meta, :cookie_banner_data
 
       def initialize(local_assigns)
+        @footer_links_count = generate_ga4_footer_links_count(local_assigns)
         @footer_navigation = local_assigns[:footer_navigation] || navigation_link_generation_from_locale(I18n.t("components.layout_footer.navigation_links"))
-        @footer_meta = local_assigns[:footer_meta] || { items: add_data_attributes_to_links(FOOTER_META[:items], "supportLink") }
+        @footer_meta = local_assigns[:footer_meta] || { items: add_data_attributes_to_links(FOOTER_META[:items], "supportLink", { section: "Support links", index_section: 3 }) }
         @cookie_banner_data = local_assigns[:cookie_banner_data] || {}
       end
 
+      def generate_ga4_footer_links_count(local_assigns)
+        footer_links_count = 0
+
+        footer_nav = local_assigns[:footer_navigation] || I18n.t("components.layout_footer.navigation_links")
+
+        footer_nav.each do |column|
+          if column[:items]
+            footer_links_count += column[:items].length
+          elsif column[:menu_contents]
+            footer_links_count += column[:menu_contents].length
+          end
+        end
+
+        footer_meta = local_assigns[:footer_meta] || FOOTER_META
+        footer_links_count += footer_meta[:items].length
+
+        extra_footer_links = 2 # Amount of links to add to @footer_links_count. This accounts for the OGL link and Crown Copyright link.
+
+        footer_links_count += extra_footer_links
+        footer_links_count
+      end
+
+      def get_footer_links_count
+        @footer_links_count
+      end
```

Comment:
I don't think you need to have this method to return `@footer_links_count`. If you want to access an instance variable like this from outside the class, you can use `attr_reader` e.g. https://github.com/alphagov/govuk_publishing_components/blob/main/lib/govuk_publishing_components/presenters/shared_helper.rb#L4